sagemaker/launch/launch_hyperparameter_tuning.py

```diff
@@ -0,0 +1,257 @@
+                hyperparameter_ranges[param_name] = ContinuousParameter(
+                    config["MinValue"],
+                    config["MaxValue"],
+                    config.get("ScalingType", "Auto"),
+                )
+        elif params_type == "IntegerParameterRanges":
+            for config in config_list:
+                param_name = config["Name"]
+                hyperparameter_ranges[param_name] = IntegerParameter(
+                    config["MinValue"],
+                    config["MaxValue"],
+                    config.get("ScalingType", "Auto"),
+                )
+        elif params_type == "CategoricalParameterRanges":
+            for config in config_list:
+                param_name = config["Name"]
+                hyperparameter_ranges[param_name] = CategoricalParameter(
+                    config["Values"]
+                )
+        else:
+            raise ValueError(
+                f"Unknown parameter type {params_type}. "
+                "Expect one of 'CategoricalParameterRanges', 'ContinuousParameterRanges', "
+                "'IntegerParameterRanges'"
+            )
+    return hyperparameter_ranges
+
+
+def run_hyperparameter_tuning_job(args, image, unknownargs):
+    """Run hyperparameter tuning job using SageMaker HyperparameterTuner"""
+
+    container_image_uri = image
+
+    prefix = f"gs-hpo-{args.graph_name}"
+
+    params = {
+        "eval-metric": args.metric_name,
+        "graph-data-s3": args.graph_data_s3,
+        "graph-name": args.graph_name,
+        "log-level": args.log_level,
+        "task-type": args.task_type,
+        "train-yaml-s3": args.yaml_s3,
+    }
+    if args.custom_script is not None:
+        params["custom-script"] = args.custom_script
+    if args.model_checkpoint_to_load is not None:
+        params["model-checkpoint-to-load"] = args.model_checkpoint_to_load
+
+    unknown_args_dict = parse_unknown_gs_args(unknownargs)
+    params.update(unknown_args_dict)
+
+    print(f"SageMaker launch parameters {params}")
+    print(f"GraphStorm forwarded parameters {unknown_args_dict}")
+
+    estimator_kwargs = parse_estimator_kwargs(args.sm_estimator_parameters)
+
+    est = PyTorch(
+        entry_point=os.path.basename(args.entry_point),
+        source_dir=os.path.dirname(args.entry_point),
+        image_uri=container_image_uri,
+        role=args.role,
+        instance_count=args.instance_count,
+        instance_type=args.instance_type,
+        output_path=args.model_artifact_s3,
+        py_version="py3",
+        base_job_name=prefix,
+        hyperparameters=params,
+        tags=[
+            {"Key": "GraphStorm", "Value": "oss"},
+            {"Key": "GraphStorm_Task", "Value": "HPO"},
+        ],
+        **estimator_kwargs,
+    )
+
+    hyperparameter_ranges = parse_hyperparameter_ranges(args.hyperparameter_ranges)
+
+    # Construct the full metric name based on user input
+    full_metric_name = f"best_{args.metric_dataset}_score:{args.metric_name}"
```

Comment:
Can you double-check whether `args.metric_dataset` and `args.metric_name` is case-sensitive and whether we should deal with it?

Reply:
Like if the user provides `Accuracy` right? I'll add `lower()` here